aws/resource_aws_ses_domain_identity_verification.go

```diff
@@ -0,0 +1,117 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/ses"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsSesDomainIdentityVerification() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSesDomainIdentityVerificationCreate,
+		Read:   resourceAwsSesDomainIdentityVerificationRead,
+		Delete: resourceAwsSesDomainIdentityVerificationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(45 * time.Minute),
+		},
+	}
+}
+
+func getAwsSesIdentityVerificationAttributes(conn *ses.SES, domainName string) (*ses.IdentityVerificationAttributes, error) {
+	input := &ses.GetIdentityVerificationAttributesInput{
+		Identities: []*string{
+			aws.String(domainName),
+		},
+	}
+
+	response, err := conn.GetIdentityVerificationAttributes(input)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting identity verification attributes: %s", err)
+	}
+
+	return response.VerificationAttributes[domainName], nil
+}
+
+func resourceAwsSesDomainIdentityVerificationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sesConn
+	domainName := strings.TrimSuffix(d.Get("domain").(string), ".")
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		att, err := getAwsSesIdentityVerificationAttributes(conn, domainName)
+		if err != nil {
+			return resource.NonRetryableError(fmt.Errorf("Error getting identity verification attributes: %s", err))
+		}
+
+		if att == nil {
+			return resource.NonRetryableError(fmt.Errorf("SES Domain Identity %s not found in AWS", domainName))
+		}
+
+		if *att.VerificationStatus != "Success" {
```

Comment:
To prevent a panic in the scenario that `att.VerificationStatus` is `nil`, we should wrap this in `aws.StringValue(att.VerificationStatus)`